### ISSUE TITLE
singleton:1 should behave like comp:1

### ISSUE THREAD

**Issue by trapd00r** [beetbox/beets#4000]
### Problem

Since singleton is a binary flag similar to comp, it should return results the same way comp does. As of now, they do differ:

```
% beet ls comp:1|wc -l
15163

% beet ls comp:true|wc -l
15163

% beet ls singleton:1|wc -l    
0

% beet ls singleton:true|wc -l
14345

**Comment by sampsyo** [comment]
Good point—what's missing here, FWIW, is a dbcore type declaration for the `singleton` field. We don't yet have type declarations for the two built-in "computed" fields on `Item`: namely, `singleton` and `filesize`. It hopefully shouldn't be too hard to add these declarations.
